src/screenshot.c

```diff
@@ -206,17 +206,23 @@ handle_screenshot_in_thread_func (GTask *task,
   gboolean interactive;
   const char *parent_window;
   const char *app_id;
+  gboolean store_permission;
 
   REQUEST_AUTOLOCK (request);
 
   app_id = xdp_app_info_get_id (request->app_info);
   parent_window = ((const char *)g_object_get_data (G_OBJECT (request), "parent-window"));
   options = ((GVariant *)g_object_get_data (G_OBJECT (request), "options"));
 
+  /* Only memorize the permission for applications that can be uniquely
+   * identified by their app ID.
+   */
+  store_permission = g_strcmp0 (app_id, "") != 0;
+
   if (xdp_dbus_impl_screenshot_get_version (impl) < 2)
     goto query_impl;
 
-  permission = xdp_get_permission_sync (app_id, PERMISSION_TABLE, PERMISSION_ID);
+  permission = store_permission ? xdp_get_permission_sync (app_id, PERMISSION_TABLE, PERMISSION_ID) : XDP_PERMISSION_UNSET;
```

Comment:
This is a upgrade path, so that if we previously had stored a permission for `app_id=""` we will stop honouring it from now on.